clients/python-wrapper/lakefs/import_manager.py

```diff
@@ -85,6 +85,8 @@ def start(self) -> str:
         """
         if self._in_progress:
             raise ImportManagerException("Import in progress")
+        if self._import_id is not None:
```

Comment:
If it's a single-use object used line this, why isn't it a function?

Reply:
AFAIK the Pythonic way (and also in many other cases) of keeping state for async execution is to return a state object that holds the coro, as for instance in [asyncio.create_task](https://docs.python.org/3/library/asyncio-task.html#creating-tasks) does.  In fact an import could be a Task, and I imagine will end up at least duck-typing like a Task.

A good way to achieve fluidity but only before calling a "start" method is with the builder pattern: you _build_ a descriptor for the executor, and call `builder.something(...).add_another_thing(...).start()`.  That returns a description of whatever is being performed.

AFAICT each important method of your single class can either only be called _before_ execution of _after_ execution.  That's why ImportManager needs a run-time state.  If we split `ImportManager` into, say, `ImportBuilder` and `ImportTask` (names TBD), then each method would go into a exactly one of these classes.  This is OO "proof" that these are two distinct objects.

Reply:
That's a valid point and we can perhaps refactor this in the future. I'm deferring it because it's requires changes in the implementation